hal/src/boron/cellular_hal.cpp

```diff
@@ -221,6 +223,34 @@ int cellular_credentials_clear(void* reserved) {
     return 0;
 }
 
+cellular_result_t cellular_global_identity(CellularGlobalIdentity* cgi_, void* reserved_) {
+    cellular_result_t result;
+    CellularGlobalIdentity cgi;
+
+    // Acquire Cellular NCP Client
+    const auto mgr = cellularNetworkManager();
+    CHECK_TRUE(mgr, SYSTEM_ERROR_UNKNOWN);
+    const auto client = mgr->ncpClient();
+    CHECK_TRUE(client, SYSTEM_ERROR_UNKNOWN);
+
+    // Validate Argument(s)
+    (void)reserved_;
+    CHECK_TRUE((nullptr != cgi_), SYSTEM_ERROR_INVALID_ARGUMENT);
+
+    // Load cached data into result struct
+    CHECK(client->getCellularGlobalIdentity(&cgi));
+
+    // Validate cache
+    CHECK_TRUE((0 != cgi.mobile_country_code && 0 != cgi.mobile_network_code &&
```

Comment:
Nitpicking (non-actionable): Prefer simple ifs for complex statements.

Reply:
Given that the expression is a conjunction, we could test each part of it individually. This would also give clearer logging so we can see which expression failed. 

```suggestion
    CHECK_TRUE((0 != cgi.mobile_country_code, SYSTEM_ERROR_BAD_DATA);
    CHECK_TRUE(0 != cgi.mobile_network_code, SYSTEM_ERROR_BAD_DATA);
    CHECK_TRUE(0xFFFF != cgi.location_area_code, SYSTEM_ERROR_BAD_DATA);
    CHECK_TRUE(0xFFFFFFFF != cgi.cell_id, SYSTEM_ERROR_BAD_DATA);
```

Reply:
eh, not sure how I make the suggestion delete more lines...